File: python/selection_sort.py
# Code originally forked from https://github.com/diptangsu/Sorting-Algorithms/

import time
from constants import *

from typing import List, TypeVar

T = TypeVar("T")


def selection_sort(array: List[T]) -> None:
    for i in range(len(array)):
        minimal = i

        for j in range(i + 1, len(array)):
            if array[minimal] > array[j]:
                minimal = j

        array[i], array[minimal] = array[minimal], array[i]


if __name__ == '__main__':
    for z in range(50):
        set_array = set2()
        start = time.time() * 1000
        selection_sort(set_array)
        end = time.time() * 1000
        elapsed = end - start
        print(elapsed)

File: rust/src/selection_sort.rs
// Code originally forked from https://github.com/diptangsu/Sorting-Algorithms/
include!("constants.rs");
use std::time::Instant;

pub fn selection_sort<T: Ord>(array: &mut [T]) {
    let length = array.len();
    for left in 0..length {
        let mut minimal = left;
        for right in (left + 1)..length {
            if array[right] < array[minimal] {
                minimal = right;
            }
        }
        array.swap(minimal, left);
    }
}

fn main() {
    unsafe {
        for _x in 0..50 {
            let start = Instant::now();
            selection_sort(&mut SET2);
            let elapsed = start.elapsed();
            println!("{:?}", elapsed);
        }
    }
}